File: rot13/src/main.js
function rot13oneLetter(letter) {
  let isUpper = letter.toLowerCase() !== letter;
  let alphabet = 'abcdefghijklmnopqrstuvwxyz';
  let index = alphabet.indexOf(letter.toLowerCase());
  if (index === -1) {
    return letter;
  }
  let newLetterIndex = (index + 13) % 26;
  let newLetter = alphabet[newLetterIndex];
  return isUpper ? newLetter.toUpperCase() : newLetter ;
}

function rot13(message){
  return message.split('').map(rot13oneLetter).join('');
}

File: rot13/src/main.py
from string import ascii_lowercase, ascii_uppercase

def rot13_char(c):
    if c.isalpha():
        alphabet = ascii_uppercase if c.isupper() else ascii_lowercase
        return alphabet[(alphabet.find(c) + 13) % 26]
    else:
        return c

def rot13(message):
    return ''.join([rot13_char(c) for c in message])